File: Docs/Stripper/cpp/stripper.cpp
// $Id: stripper.cpp,v 1.1.1.1 2001/07/17 22:59:22 micah Exp $
//
// This program implements a C++ comment stripper
// using the State Map Parser.
//
#include <iostream>
#include <stdlib.h>
#include "stripFSM.h"

using namespace std;

int main()
{
	smc::StripFSM myStripper;
	myStripper.SetStreams(cin,cout);

	while (cin)
		switch(myStripper.ReadChar())
		{
		  case EOF: exit(0); break;
			case '/': myStripper.Slash(); break;
			case '*': myStripper.Star(); break;
			case '\n': myStripper.EOL(); break;
			default: myStripper.Other(); break;
		}
}

File: Docs/Stripper/java/Stripper.java
package smc.stripper;

import java.io.InputStream;
import java.io.OutputStream;
import java.io.IOException;

import smc.stripper.StripFSM;

class Stripper
{
    public static void main(String args[]) 
    {
        StripFSM s = new StripFSM();
        s.SetStreams( System.in, System.out);
        try
        {
            while( true )
            {
                int c = s.ReadChar();
                switch( c )
                {
                    case -1: System.exit(0); break; 
		    	    case '/': s.Slash(); break;
		    	    case '*': s.Star(); break;
		    	    case '\n': s.EOL(); break;
		    	    default: s.Other(); break;
		        }
            }
        }
        catch (Exception e )
        {  
            System.out.println();
            System.out.println(e);
            System.err.println(e); 
        }
    }
     
}
